nodejs/awsx/aws_test.go

```diff
@@ -250,54 +313,70 @@ func Test_Examples(t *testing.T) {
-		// 	StackName: addRandomSuffix("ec2"),
-		// 	Config: map[string]string{
-		// 		"aws:region":               region,
-		// 		"cloud:provider":           "aws",
-		// 		"containers:redisPassword": "SECRETPASSWORD",
-		// 	},
-		// 	Dependencies: []string{
-		// 		"@pulumi/awsx",
-		// 	},
-		// 	Quick:       true,
-		// 	SkipRefresh: true,
-		// 	PreviewCommandlineFlags: []string{
-		// 		"--diff",
-		// 	},
-		// 	EditDirs: []integration.EditDir{
-		// 		{
-		// 			Additive: true,
-		// 			Dir:      path.Join(cwd, "../examples/ec2/update1"),
-		// 		},
-		// 		{
-		// 			Additive:               true,
-		// 			Dir:                    path.Join(cwd, "../examples/ec2/update2"),
-		// 			ExtraRuntimeValidation: containersRuntimeValidator(region, false /*isFargate*/),
-		// 		},
-		// 	},
-		// },
+		})
+
+	integration.ProgramTest(t, &test)
+}
+
+func skipIfShort(t *testing.T) {
+	if testing.Short() {
+		t.Skip("skipping long-running test in short mode")
 	}
+}
 
-	tests := shortTests
-	if !testing.Short() {
-		tests = append(tests, longTests...)
+func getEnvRegion(t *testing.T) string {
+	envRegion := os.Getenv("AWS_REGION")
+	if envRegion == "" {
+		t.Skipf("Skipping test due to missing AWS_REGION environment variable")
 	}
+	fmt.Printf("AWS Region: %v\n", envRegion)
 
-	for _, ex := range tests {
-		example := ex.With(integration.ProgramTestOptions{
-			ReportStats: integration.NewS3Reporter("us-west-2", "eng.pulumi.com", "testreports"),
-			// TODO[pulumi/pulumi#1900]: This should be the default value, every test we have causes some sort of
-			// change during a `pulumi refresh` for reasons outside our control.
-			ExpectRefreshChanges: true,
-		})
-		t.Run(example.Dir, func(t *testing.T) {
-			integration.ProgramTest(t, &example)
-		})
+	return envRegion
+}
+
+func getCwd(t *testing.T) string {
+	cwd, err := os.Getwd()
+	if err != nil {
+		t.FailNow()
 	}
+
+	return cwd
+}
+
+func getBaseOptions(t *testing.T) integration.ProgramTestOptions {
+	envRegion := getEnvRegion(t)
+	baseJS := integration.ProgramTestOptions{
+		Config: map[string]string{
+			"aws:envRegion": envRegion,
+		},
+		Dependencies: []string{
+			"@pulumi/awsx",
+		},
+		Quick:       true,
+		SkipRefresh: true,
+		// TODO[pulumi/pulumi#1900]: This should be the default value, every test we have causes some sort of
+		// change during a `pulumi refresh` for reasons outside our control.
+		ExpectRefreshChanges: true,
+		ReportStats:          integration.NewS3Reporter("us-west-2", "eng.pulumi.com", "testreports"),
+	}
+
+	return baseJS
+}
+
+func getDefaultProviderOptions(t *testing.T) integration.ProgramTestOptions {
+	baseJS := integration.ProgramTestOptions{
+		Dependencies: []string{
+			"@pulumi/awsx",
+		},
+		Quick:       true,
+		SkipRefresh: true,
+		// TODO[pulumi/pulumi#1900]: This should be the default value, every test we have causes some sort of
+		// change during a `pulumi refresh` for reasons outside our control.
```

Comment:
JFWIW - We decided to close 1900 as by design, so we probably can just remove this comment.